backend/src/test/java/com/github/ingbeck/backend/controller/DiaryControllerTest.java

```diff
@@ -34,4 +45,104 @@ void createNewDiary_whenCalledWithUserId_thenReturnNewDiaryWithUserId() throws E
                 .andExpect(jsonPath("$.id").isNotEmpty());
     }
 
+    @Test
+    void updateDiaryEntry_whenCalledWithEmptyDiaryEntry_addNewDiaryEntry() throws Exception{
+        //GIVEN
+        repository.save(new Diary("1", "1", List.of()));
+
+        //WHEN & THEN
+        mvc.perform(put("/api/diaries/1/2024-04-11")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content("""
+                        [
+                              {
+                                "name": "Kinderriegel",
+                                "amount": 1,
+                                "unit": "",
+                                "calories": 54,
+                                "mealType": "DINNER"
+                              }
+                         ]
+                         """))
+                .andExpect(status().isOk())
+                .andExpect(content()
+                        .json("""
```

Comment:
I recommend objectMapper. In my opinion it improves the readability a lot.
Example from my code:
```
AppUserRegister userRegister = new AppUserRegister("user", "password");
MvcResult userJson = mvc.perform(post("/api/users/register")
                        .contentType(MediaType.APPLICATION_JSON)
                        .content(objectMapper.writeValueAsString(userRegister)))
                .andExpect(status().isCreated())
                .andReturn();
AppUser user = objectMapper.readValue(userJson.getResponse().getContentAsString(), AppUser.class)
```
In this example a user gets created and returned. Afterwards I got the returned user and can check all the things I might want to check.

Reply:
THX. I will consider to refactor my tests.